client/src/components/Users.jsx

```diff
@@ -1,64 +1,55 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import EntityTypeHeader from './entity-editor/EntityTypeHeader';
+import ConfirmUserAction from './modals/ConfirmUserAction';
 import InviteUser from './modals/InviteUser';
 import * as api from '../api';
 
 require('../styles/EntityTypeHeader.scss');
 require('../styles/Users.scss');
 
-class UserActionSelector extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { action: '?' };
-    this.onChange = this.onChange.bind(this);
-  }
-
-  onChange(event) {
-    const action = event.target.value;
-    const userId = this.props.userId;
-    this.setState({ action });
-    this.props.onChange(userId, action);
-  }
-
-  render() {
-    const { active, admin } = this.props;
-    return (
-      <select
-        className="UserActionSelector"
-        onChange={this.onChange}
-        value={this.state.action}
-      >
-        <option value="?">...</option>
-        <option disabled key="user-edit" value="edit">
-          Edit
-        </option>
-        <option disabled={active} key="user-delete" value="delete">
-          Delete
-        </option>
-        {!admin
-           ?
-             <option disabled={admin} key="user-promote" value="promote">
-               Enable admin privileges
-             </option>
-           :
-             <option disabled={(!admin || active)} key="user-demote" value="demote">
-               Remove admin privileges
-             </option>
-        }
-      </select>
-    );
-  }
+function UserActionSelector({ user, onChange }) {
+  const { active, admin } = user;
+  return (
+    <select
+      className="UserActionSelector"
+      onChange={event => onChange(user, event.target.value)}
+      value="?"
+    >
+      <option value="?">...</option>
+      <option disabled key="user-edit" value="edit">
```

Comment:
I think I saw a console warning about flattened children (!) - but it's probably harmless.